infra/testing/sw-env-mocks/Headers.js

```diff
@@ -18,7 +18,7 @@
 // https://fetch.spec.whatwg.org/#headers-class
 class Headers {
   constructor(obj = {}) {
-    this.obj = obj;
+    this.obj = Object.assign({}, obj);
```

Comment:
This works around an issue that I encountered in the tests.

https://github.com/pinterest/service-workers/blob/master/packages/service-worker-mock/models/Headers.js looks like it's much more robust than it used to be, and perhaps we should migrate off of our own mock at some point in the future.